# test-pca
*Date: 1/19/2024*

A vignette for running PCA from scratch for Einstein Omics Club to run on the HPC. Adapted from Kevin Blighe: [Biostars Post](https://www.biostars.org/p/335605/)

## Program requirements:
- plink v1.9
- plink2 v2.0
- jupyter 1.0.0
- python v3.12.1
- bcftools (tested on v1.10)

*Downloaded data (VCF.gz and tab-indices), ~ 15.5 GB converted BCF files and their indices, ~14 GB binary PLINK files, ~53 GB pruned PLINK binary files, ~ <1 GB*

## Introduction to the Vignette
The goal of this tutorial is to learn how to analyze genetic variants to understand how a population is structured. The tools we will use in this tutorial are commonly used analysis programs in the field of population genetics that have been applied to studying human genetic diversity across many populations. The standard data type used for these genetic ancestry analyses are single nucleotide polymorphisms (SNPs) generated by genotyping or whole-genome sequencing (WGS) human biological samples.

*Warning from Joe Pickrell:* "The tools you will run in this tutorial provide summaries of the data that are informative about this question. However, be aware that there is no black box solution--all methods can be misleading, and the main difficulty is not running software, but instead interpreting the results."

## Dataset Description (1KG Phase 3)
We will be analyzing SNPs from the [1000 Genomes Project (Phase 3)](https://www.nature.com/articles/nature15393)
- 2,504 individuals from 26 populations
- 84,700,000 SNPs from low coverage WGS, deep exome sequencing, and dense microarray genotyping.
- 3,600,000 small insertions/deletions (indels)
- 60,000 structural variants
- Haplotype phased
- >99% of these variants occur at >1% frequency for multiple populations.
- We will be using the integrated phased dataset for unrelated samples (aligned to GRCh37)

# Analysis Pipeline Overview
Follow the steps in data_processing.md

1) Download the dataset.(1000KG SNPs)
2) Download the reference genome (GRCh37)
3) Convert to bcf file format
4) Convert to PLINK format
5) Prune variants for ancestry infomrative markers 
6) Merge across all chromosomes
7) Run Principal Components Analysis (PCA)


## Pre-step 1) Prepare conda environment
```bash
conda create --name ancestry_vignette
conda activate ancestry_vignette
conda install -c bioconda bcftools plink plink2
bcftools --version
plink --version
plink2 --version
```

# Step 1) Download files 
Download variants (*.vcf) and accompanying files into their repsected directories:

sbatch 1_get-data.sh

# Step 2) Download the GRCh37 reference genome

sbatch 2_get-ref.sh

# Step 3) Convert the 1KG dataset into a BCF for easier handling

sbatch 3_para-convert2bcf.sh

- Ensure that multi-allelic calls are split and that indels are left-aligned compared to reference genome (1st pipe)
- Sets the ID field to a unique value: CHROM:POS:REF:ALT (2nd pipe)
- Removes duplicates (3rd pipe)
- -I +'%CHROM:%POS:%REF:%ALT' means that unset IDs will be set to CHROM:POS:REF:ALT*
- -x ID -I +'%CHROM:%POS:%REF:%ALT' first erases the current ID and then sets it to CHROM:POS:REF:ALT*

*Continuous error for chr2 (below):

Processing chromosome 2
Normalizing and re-annotating variants...
[E::bgzf_uncompress] Inflate operation failed: 3
Lines   total/split/joined/realigned/skipped:   432433/2320/0/578/0
Lines   total/split/joined/realigned/skipped:   434860/0/0/0/0
Indexing the BCF file...
Chromosome 2 processing completed.

# Step 4) Convert the BCF files into PLINK format

sbatch 4_convert2PLINK.sh

# Step 5) Prune variants from each chromosome ("LD pruning")
Linkage disequilibrium (LD) refers to the non-random association of alleles at different loci in a given population. LD pruning, or linkage disequilibrium pruning, is a process used in population genetics and genetic association studies to reduce the redundancy of genetic information by selecting a subset of variants that are less correlated. 

For this exercise, we will prune variants to reduce redundancy and improve computational efficiency.

sbatch 5_LD-prune.sh

- --maf 0.10, only retain SNPs with MAF greater than 10%
- --indep [window size] [step size/variant count)] [Variance inflation factor (VIF) threshold]

How many variants remain?

# Step 6) Merge all chromosomes into a single plink file

sbatch 6_merge.sh

# Step 7) Run Principal Components Analysis (PCA)

sbatch 7_pca.sh

# Step 8) Plot PCA Results

# Step 9) Run Adixture analysis 